slides/aml-25-042518-time-series/index.html

```diff
@@ -0,0 +1,952 @@
+```
+]
+
+.center[![:scale 65%](images/autocorr 3.png)]
+
+???
+Note - 1 slide split into 3 slides (2)
+
+---
+
+.smaller[
+```python
+from statsmodels.tsa.stattools import acf
+autocorrelation = acf(ppm.diff()[1:])
+plt.plot(autocorrelation)
+```
+]
+
+.center[![:scale 65%](images/autocorr 4.png)]
+
+???
+Note - 1 slide split into 3 slides (3)
+
+---
+class: compact
+
+.smaller[
+- Seasonal model for CO2
+```python
+from statsmodels.tsa.seasonal import seasonal_decompose
+decomposition = seasonal_decompose(ppm, model='additive')
+fig = decomposition.plot()
+#fig.set_figheight(6)
+fig.tight_layout()
+```
+]
+
+.center[![:scale 55%](images/seasonal model.png)]
+
+
+???
+
+---
+class: spacious
+
+# Autoregressive (linear) model
+
+- Order k AR model (can include trend and offset):
+
+`$$ x_{t+k} = c_0  x_t + c_1 x_{t + 1} + ... + c_{k - 1} x_t $$`
+
+- learn $c_i$ (i.e. using least squares)
+
+???
+
+---
+
+# AR model with statsmodels
+
+.left-column[
+.smaller[
+```python
+from statsmodels.tsa import ar_model
+ar = ar_model.AR(ppm[:500])
+res = ar.fit(maxlag=12)
+res.params
+```
+```
+const              -2.018779
+L1.interpolated     0.931758
+L2.interpolated    -0.240629
+L3.interpolated    -0.099048
+L4.interpolated    -0.076488
+L5.interpolated     0.164472
+L6.interpolated    -0.000531
+L7.interpolated    -0.023481
+L8.interpolated    -0.125134
+L9.interpolated     0.100288
+L10.interpolated   -0.021266
+L11.interpolated    0.414740
+L12.interpolated   -0.017268
+dtype: float64
+```
+]
+]
+
+.right-column[
+.smaller[
+```python
+ppm[300:].plot(label="true")
+res.predict(ppm.index[500], 
+  ppm.index[-1]).plot(label="predict")
+plt.legend(loc="best")
+```
+]
+.center[![:scale 100%](images/ar model with statsmodel.png)]
+]
+
+???
+
+---
+
+# Impact of order
+
+.left-column[
+.smaller[
+```python
+ar6 = ar_model.AR(ppm[:500])
+res = ar6.fit(maxlag=6)
+ar_pred = res.predict(ppm.index[500], 
+                      ppm.index[-1])
+ppm[300:].plot(label="true")
+ar_pred.plot(label="AR predict")
+plt.legend(loc="best")
+```
+]
+.center[![:scale 90%](images/impact of order 1.png)]
+]
+
+.right-column[
+.smaller[
+```python
+ar25 = ar_model.AR(ppm[:500])
+res = ar25.fit(maxlag=25)
+ar_pred = res.predict(ppm.index[500], 
+                      ppm.index[-1])
+ppm[300:].plot(label="true")
+ar_pred.plot(label="AR predict")
+plt.legend(loc="best")
+```
+]
+.center[![:scale 90%](images/impact of order 2.png)]
+]
+
+???
+
+---
+
+# ARIMA
+
+.smaller[
+```python
+from statsmodels import tsa
+arima_model = tsa.arima_model.ARIMA (ppm[:500], order=(12, 1, 0))
+res = arima_model.fit()
+arima_pred = res.predict(ppm.index[500], ppm.index[-1], typ="levels")
+ppm[300:].plot(label="true")
+arima_pred.plot(label="ARIMA predict")
+plt.legend(loc="best")
+```
+]
+
+.left-column[
+
+.center[![:scale 90%](images/arima.png)]
+
+]
+
+.right-column[
+
+- More complicated process model
+- Internally uses differencing
+- Also checkout SARIMAX (state space model)
+
+]
+
+???
+
```

Comment:
can you make a note about specifying type='levels' in predict? (was highlighted on the slide)

Reply:
done